ixmp4/data/db/optimization/column/model.py

```diff
@@ -0,0 +1,38 @@
+from typing import ClassVar
+
+from sqlalchemy import UniqueConstraint
+
+from ixmp4 import db
+from ixmp4.data import types
+from ixmp4.data.abstract import optimization as abstract
+from ixmp4.data.db.optimization.indexset import IndexSet
+
+from .. import base
+
+
+class Column(base.BaseModel):
+    NotFound: ClassVar = abstract.Column.NotFound
+    NotUnique: ClassVar = abstract.Column.NotUnique
+    DeletionPrevented: ClassVar = abstract.Column.DeletionPrevented
+
+    name: types.String = db.Column(db.String(255), nullable=False, unique=False)
+    # Currently not in use:
+    dtype: types.String = db.Column(
+        db.String(255), nullable=False, unique=False
+    )  # pandas dtype
+
+    table__id: types.Mapped[int] = db.Column(
+        db.Integer, db.ForeignKey("optimization_table.id"), index=True
+    )
+    indexset: types.Mapped[IndexSet] = db.relationship(single_parent=True)
+    constrained_to_indexset: types.Mapped[int] = db.Column(
+        db.Integer, db.ForeignKey("optimization_indexset.id"), index=True
+    )
+
+    # Currently not in use:
+    unique: types.Boolean = db.Column(db.Boolean, default=True)
+
+    __table_args__ = (UniqueConstraint(name, table__id, constrained_to_indexset),)
```

Comment:
Can there be two columns with the same names but constrained to different sets?
Not really, right? Probably need to adjust this constraint.

Reply:
Yes, @danielhuppmann mentioned above that only the names should be unique, so I'll drop `constrained_to_indexsets` from this `UniqueConstraint`.